app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -0,0 +1,262 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Heatmap from "~/components/visualizations/heatmap/Heatmap";
+import { getAMRCounts } from "~/api/amr";
+
+import cs from "./amr_heatmap_vis.scss";
+
+const METRIC_ALLELES = "alleles";
+const METRIC_GENES = "genes";
+
+export default class AMRHeatmapVis extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      viewLevel: "genes",
+      metric: "coverage",
+    };
+
+    this.heatmap = null;
+  }
+
+  componentDidMount() {
+    this.requestAMRCountsData(this.props.sampleIds);
+  }
+
+  componentDidUpdate() {
+    let rows = this.createHeatmapLabels();
+    let columns = this.state.samples;
+    let values = this.computeHeatmapValues(rows);
+    let options = {};
+    this.renderHeatmap(rows, columns, values, options);
```

Comment:
I don't think you need to define `options` here and then pass it in. If you do end up having options, you can just put them inside the `renderHeatmap` function.